### PR TITLE
[mongodb-queue] Updated add function parameter type

### PR DESCRIPTION
**Pull request by codejockie**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/chilts/mongodb-queue#add
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongodb-queue/index.d.ts [hunk 1]
```diff
@@ -13,8 +14,8 @@ declare namespace mongodbQueue {
         constructor(db: Db, name: string, opts?: QueueOptions);
 
         createIndexes(callback: QueueCallback<string>): void;
-        add(payload: string, callback: QueueCallback<string>): void;
-        add(payload: string, opts: QueueOptions, callback: QueueCallback<string>): void;
+        add(payload: Payload, callback: QueueCallback<string>): void;
```
Comment (FiveOFive) [inline types/mongodb-queue/index.d.ts:17]: Looks like when the payload is an array, the callback returns an array of ids. I think two signatures `        add(payload: Payload, callback: QueueCallback<string>): void;` and `add(payload: Payload[], callback: QueueCallback<string[]>): void;` might be best. And then the Payload type would just be `type Payload = string | Record<string, unknown>;`

**Review by FiveOFive — CHANGES_REQUESTED** [review]
Nice update!

I think the QueueMessage interface also needs to be updated from payload: string to payload: Payload. And one small suggestion for add one message vs add an array of messages.

File: types/mongodb-queue/index.d.ts (hunk 1 above)
Comment (codejockie) [reply types/mongodb-queue/index.d.ts:17]: Yeah right, except type Payload would still need to maintain the Array part for `Payload[]`.

Comment (FiveOFive) [reply types/mongodb-queue/index.d.ts:17]: Right, makes sense!

Comment (codejockie) [reply types/mongodb-queue/index.d.ts:17]: I just updated the code.

**Review by FiveOFive — APPROVED** [review]

**Comment by codejockie** [comment]
Ready to merge

**Pull request merged**